lib/validators/omnivore.js

```diff
@@ -38,7 +40,56 @@ module.exports = function validateOmnivore(opts, callback) {
       if (size > limits.max_filesize) {
         return callback(invalid('File is larger than ' + prettyBytes(limits.max_filesize) + '. Too big to process.'));
       }
-      callback();
+      //only look at files that passed above file size test
+      files.forEach(function(f) {
+        q.defer(validateGdal, f, filetype);
```

Comment:
See below -- you shouldn't have to use an async queue here since the function is really synchronous. 

Also, the only case in which there would be more than one file is the shapefile case. Maybe just treat that explicitly instead of looping through all the files?